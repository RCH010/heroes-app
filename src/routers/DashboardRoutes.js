import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { DcScreen } from '../components/dc/DcScreen'
import { HeroScreen } from '../components/heroes/HeroScreen'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { SearchScreen } from '../components/search/SearchScreen'
import { Navbar } from '../components/ui/Navbar'

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar/>
            <div className='container-fluid'>
                <Switch>
                    <Route exact path='/marvel' component= { MarvelScreen} />
                    <Route exact path='/dc' component= { DcScreen } />
                    <Route exact path='/hero/:heroeId' component= { HeroScreen } />
                    <Route exact path='/search' component= {SearchScreen} />
                    <Redirect to ='/marvel'/>
                </Switch>
            </div>
        </>
    )
}
