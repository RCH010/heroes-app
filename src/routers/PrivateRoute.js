import React from 'react'
import { Redirect, Route } from 'react-router-dom'
import PropTypes from 'prop-types';

export const PrivateRoute = ({
    isAuthenticated,
    component: Component,
    ...rest
}) => {
    // recordar la ruta
    localStorage.setItem('lastPath', rest.location.pathname);

    return (
        
        <Route {...rest}
            component = { (props) => (
                (isAuthenticated) 
                    ? <Component {...props} />
                    : <Redirect to='/login' />
            )}
        />
    )
}

PrivateRoute.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}