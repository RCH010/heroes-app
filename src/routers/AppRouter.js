import React, { useContext } from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
} from "react-router-dom";
import { AuthContext } from '../auth/AuthContext';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

    const {user} = useContext(AuthContext)

    return (
        <Router>
            <div>
                <Switch>
                    <PublicRoute 
                        exact
                        component={LoginScreen}
                        isAuthenticated={user.logged}
                        path='/login'
                    />
                    {/* <Route exact path='/login' component={LoginScreen} /> */}
                    <PrivateRoute 
                        path='/' 
                        component={DashboardRoutes}
                        isAuthenticated={user.logged}
                    />
                    {/* Aquí este AppRouter es el principal, y en el caso que no sea login
                        se va y carga el DashboardRoutes. Ahí hay un switch que es el que entraría
                        aquí... */}
                </Switch>
            </div>
        </Router>
    )
}
