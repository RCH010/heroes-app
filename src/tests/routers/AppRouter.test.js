import { mount } from "enzyme"
import { AuthContext } from "../../auth/AuthContext";
import { AppRouter } from "../../routers/AppRouter"

describe('Tests in <AppRouter />', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    };

    test('should display login if not authenticated', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );
        
        expect(wrapper).toMatchSnapshot();
    })

    test('should display marvel component if authenticated', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                name:'Raul',
                logged: true
            }
        };
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );
        
        expect(wrapper.find('.navbar').exists()).toBe(true);
    })
    
    
})
