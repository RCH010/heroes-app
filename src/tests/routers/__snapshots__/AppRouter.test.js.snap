// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AppRouter /> should display login if not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login Screen
    </h1>
    <br />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
